pkg/terraform/resource/configuration.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+// ConfigureWithNameFn functions configure the base resource fields by using
+// given name value.
+type ConfigureWithNameFn func(base map[string]interface{}, name string)
+
+// ConfigurationOption allows setting optional fields of a Configuration object.
+type ConfigurationOption func(*Configuration)
+
+// WithExternalName allows you to set an ExternalNamer for given Configuration.
+func WithExternalName(e ExternalNamer) ConfigurationOption {
+	return func(c *Configuration) {
+		c.ExternalNamer = e
+	}
+}
+
+// WithTerraformIDFieldName allows you to set TerraformIDFieldName.
+func WithTerraformIDFieldName(n string) ConfigurationOption {
+	return func(c *Configuration) {
+		c.TerraformIDFieldName = n
+	}
+}
+
+// NewConfiguration returns a new *Configuration.
+func NewConfiguration(version, kind, terraformResourceType string, opts ...ConfigurationOption) *Configuration {
+	c := &Configuration{
+		Version:               version,
+		Kind:                  kind,
+		TerraformResourceType: terraformResourceType,
+		TerraformIDFieldName:  "id",
+	}
+	for _, f := range opts {
+		f(c)
+	}
+	return c
+}
+
+// ExternalNamer contains all information that is necessary for naming operations,
+// such as removal of those fields from spec schema and calling Configure function
+// to fill attributes with information given in external name.
+type ExternalNamer struct {
```

Comment:
`ExternalNamer` sounds like it contains configuration/information to build the external name (using existing fields), whereas, it is mostly for almost the opposite, to parse the external name into spec. Could we find a more appropriate name for this type to prevent possible confusion, e.g. `NamingConfiguration` ?

Reply:
Named the type as `ExternalNameConfiguration` but kept the field name in `Configuration` as `ExternalName`